src/main/java/com/quorum/gauge/ext/EthSignTransaction.java

```diff
@@ -22,7 +22,29 @@
 import org.web3j.protocol.core.Response;
 
 import java.util.Map;
+import java.util.Optional;
 
-public class EthSignTransaction extends Response<Map<String, Object>> {
+public class EthSignTransaction extends Response<Object> {
 
+    public Optional<String> getRaw() {
+        /**
+         * Geth (at least up to v1.10.1) returns an object with raw and tx (check https://github.com/ethereum/go-ethereum/blob/9c653ff6625df1ff0f6c6612958b2a1da0021e40/internal/ethapi/api.go#L1821)
+         * However, the official spec https://eth.wiki/json-rpc/API details that eth_signTransaction returns only the hash.
+         *
+         * This code is to support both behaviours while Geth is not fixed
+         */
+
+        Optional<Object> result = Optional.ofNullable(getResult());
+
+        if (result.isEmpty()) {
+            return Optional.empty();
+        }
+
+        if (result.get() instanceof String) {
+            return Optional.of((String) result.get());
+        }
+
+        Map<String, Object> map = (Map<String, Object>) result.get();
+        return Optional.of((String) map.get("raw"));
```

Comment:
this should fix all the unable to create HashMap errors 👍

Reply:
yes, that is the goal. it will handle both scenarios 👍